fuzz_test.go

```diff
@@ -0,0 +1,90 @@
+package orderedmap
+
+// Adapted from https://github.com/dvyukov/go-fuzz-corpus/blob/c42c1b2/json/json.go
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/dvyukov/go-fuzz-corpus/fuzz"
+)
+
+func FuzzMarshalling(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		for _, ctor := range []func() any{
+			func() any { return nil },
+			func() any { return new([]any) },
+			func() any { return &OrderedMap[string, string]{} },
+			func() any { return &OrderedMap[string, any]{} },
+			func() any { return new(S) },
+		} {
+			v := ctor()
+			if json.Unmarshal(data, v) != nil {
+				continue
+			}
+			if s, ok := v.(*S); ok {
+				if len(s.P) == 0 {
+					s.P = []byte(`""`)
+				}
+			}
+			data1, err := json.Marshal(v)
+			if err != nil {
+				panic(err)
+			}
+			v1 := ctor()
+			if json.Unmarshal(data1, v1) != nil {
+				continue
+			}
+			if s, ok := v.(*S); ok {
+				// Some additional escaping happens with P.
+				s.P = nil
+				v1.(*S).P = nil
+			}
+			if !fuzz.DeepEqual(v, v1) {
```

Comment:
I'd rather not pull a whole dependency just for this, which is a light wrapper around reflect.DeepEqual

Reply:
Done :)